spec/controllers/infra_networking_controller_spec.rb

```diff
@@ -1,20 +1,65 @@
 describe InfraNetworkingController do
-  let(:zone) { FactoryGirl.build(:zone) }
-  let!(:server) { EvmSpecHelper.local_miq_server(:zone => zone) }
+  let!(:switch) { FactoryGirl.create(:switch, :name => 'test_switch1', :shared => 'true') }
+  let(:classification) { FactoryGirl.create(:classification, :name => "department", :description => "Department") }
+  let(:tag1) { FactoryGirl.create(:classification_tag, :name   => "tag1", :parent => classification) }
+  let(:tag2) { FactoryGirl.create(:classification_tag, :name   => "tag2", :parent => classification) }
+  let(:ems) { FactoryGirl.create(:ems_vmware) }
+  let(:cluster) { FactoryGirl.create(:ems_cluster, :ems_id => ems.id) }
 
-  before { stub_user(:features => :all) }
+  before do
+    stub_user(:features => :all)
+    FactoryGirl.create(:tagging, :tag => tag1.tag, :taggable => switch)
+    FactoryGirl.create(:tagging, :tag => tag2.tag, :taggable => switch)
+  end
 
-  describe '#tree_select' do
+  describe 'render_views' do
     render_views
 
-    context 'switch' do
-      let(:ems) { FactoryGirl.create(:ems_vmware) }
-      let(:cluster) { FactoryGirl.create(:ems_cluster, :ems_id => ems.id) }
-      let(:host) { FactoryGirl.create(:host_vmware, :ems_id => ems.id, :ems_cluster => cluster) }
-      let(:switch) do
-        sw = FactoryGirl.create(:switch_vmware, :ems_id => ems.id)
-        FactoryGirl.create(:host_switch, :host => host, :switch => sw)
-        sw
+    before do
+      EvmSpecHelper.create_guid_miq_server_zone
+    end
+
+    describe '#explorer' do
+      before do
+        session[:settings] = {:views => {}, :perpage => {:list => 5}}
+        session[:sb] = {:active_accord => :infra_networking_accord}
+        seed_session_trees('switch', :infra_networking_tree, 'root')
+        get :explorer
```

Comment:
This `get` shouldn't be in the before block as this is the vital part of the testing...